generate/unmarshal.go.tmpl

```diff
@@ -31,13 +31,61 @@ func (v *{{.GoName}}) UnmarshalJSON(b []byte) error {
         return err
     }
 
-    {{/* Now, for each field, call out to the unmarshal-helper. */}}
-    {{range .AbstractFields -}}
-    err = __unmarshal{{.GoType.Reference}}(
-        &v.{{.GoName}}, firstPass.{{.GoName}})
-    if err != nil {
-        return err
+    {{/* Now, for each field, call out to the unmarshal-helper.
+         This gets a little complicated because we may have a slice field.
+         So what we do is basically, for each field:
+
+            target := &v.MyField      // *[][]...[]MyType
```

Comment:
Same nit here about this not being a supported type.

Reply:
This is correct as-is -- `target` has type `*[][]...[]MyType` (or `*[][]...[]*MyType`, because it's a pointer to the field-type, again because that simplifies the templating at the cost of making the generated code look a bit strange.